build.gradle

```diff
@@ -1,37 +1,40 @@
 buildscript {
-    ext.kotlinVersion = "1.4.20"
-    ext.compileVersion = 28
+    ext.kotlinVersion = "1.6.21"
+    ext.compileVersion = 31
     ext.buildToolsVersion = "30.0.3"
     ext.minVersion = 14
     ext.billingVersion = "4.1.0"
     ext.billingClient3Version = "3.0.3"
-    ext.lifecycleVersion = "2.3.0-rc01"
+    ext.lifecycleVersion = "2.5.0-rc01"
     ext.testLibrariesVersion = "1.4.0"
     ext.testJUnitVersion  = "1.1.3"
-    ext.robolectricVersion = "4.6.1"
-    ext.mockkVersion = "1.10.0"
-    ext.assertJVersion = "3.13.2"
+    ext.robolectricVersion = "4.8.1"
+    ext.mockkVersion = "1.12.4"
+    ext.assertJVersion = "3.22.0"
     ext.annotationVersion = "1.3.0"
+    ext.androidxCoreVersion = "1.8.0"
+    ext.mockwebserverVersion = "4.2.0"
 
     repositories {
         mavenCentral()
         google()
     }
     dependencies {
         classpath "com.vanniktech:gradle-maven-publish-plugin:0.15.1"
-        classpath 'com.android.tools.build:gradle:7.0.2'
+        classpath 'com.android.tools.build:gradle:7.2.1'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion"
-        classpath "org.jetbrains.dokka:dokka-gradle-plugin:1.6.10"
+        classpath "org.jetbrains.dokka:dokka-gradle-plugin:1.6.21"
     }
 }
 
 plugins {
     id "io.gitlab.arturbosch.detekt" version "1.17.0-RC3"
     id "com.savvasdalkitsis.module-dependency-graph" version "0.9"
+    id "com.github.ben-manes.versions" version "0.42.0"
```

Comment:
I was thinking about including some sort of automation that automatically upgrades dependencies instead of this plugin. I used this one to fix a couple issues I was having but I don't think we have to merge it into main